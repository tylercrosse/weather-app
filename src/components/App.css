.recentSearches {
  margin-top: 20px;
  text-align: center;
}

.recentSearches__btn {
  background-color: #fcfcfc;
  border: 0;
}

.recentSearch__item {
  background-color: #fcfcfc;
  border: 0;
  margin: 0 20px;
  padding: 20px;
  cursor: pointer;
}

.search {
  width: 600px;
  margin: 20px auto 0;
}

.search__bar {
  position: relative;
  padding-bottom: '0px';
}

.search__input,
.search__input:focus,
.search__input:active {
  box-shadow: 0 2px 2px 0 rgba(0, 0, 0, 0.16), 0 0 0 1px rgba(0, 0, 0, 0.08);
  border: honeydew;
  display: block;
  width: 100%;
  padding: 16px;
  font-size: 16px;
  border-radius: 2px;
  outline: none;
}

.search__item {
  padding: 10px;
  color: #555;
  cursor: pointer;
  margin-top: 0;
  margin-bottom: 0;
}

.search__autocomplete_container {
  position: absolute;
  width: 100%;
  border-bottom: honeydew;
  border-left: honeydew;
  border-right: honeydew;
  border-top: 1px solid #e6e6e6;
  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.2);
  border-radius: 0 0 2px 2px;
}
