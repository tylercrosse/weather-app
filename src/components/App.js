import React, { Component } from 'react';
import { connect } from 'react-redux';
import Search from './Search';
import './App.css';

class App extends Component {
  componentDidMount() {
    console.log('locations ðŸ—º:', this.props.locations)
  }
  render() {
    return (
      <div className="App">
        <Search />
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  locations: state.locations
})

export default connect(
  mapStateToProps
)(App);
