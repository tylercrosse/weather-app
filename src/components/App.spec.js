import React from 'react';
import { shallow } from 'enzyme';
import toJson from "enzyme-to-json";
import { App } from './App';

const setup = () => {
  const props = {
    locations: {},
    weather: {},
    geocode: jest.fn(),
    fetchForecast: jest.fn(),
  }

  const component = <App {...props} />;
  const wrapper = shallow(component);

  return {
    props,
    wrapper
  }
}

describe('<App />', () => {
  it('should render correctly', () => {
    const { wrapper } = setup();
    expect(toJson(wrapper)).toMatchSnapshot();
  });
})
