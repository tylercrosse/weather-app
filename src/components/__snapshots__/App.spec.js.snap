// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render correctly 1`] = `
<div
  className="App"
>
  <section
    className="current"
  >
    <Search
      geocode={[Function]}
    />
  </section>
</div>
`;
